### PR TITLE
Issue544 exponential damper

### PR DESCRIPTION
**Pull request by dhblum**
Made the proposed changes from #544 .

### CODE REVIEW

**Comment by dhblum** [comment]
I realized I made this pull request without running full regression tests. At least Fluid,Actuators.Dampers.Examples.MixingBox breaks due to, now, missing parameters.

File: Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo [hunk 1]
```diff
@@ -3,18 +3,14 @@ partial model PartialDamperExponential
...
+      eta_default*ReC*sqrt(A)*facRouDuc);
    extends Annex60.Fluid.Actuators.BaseClasses.ActuatorSignal;
  parameter Boolean use_deltaM = true
     "Set to true to use deltaM for turbulent transition, else ReC is used";
  parameter Real deltaM = 0.3
     "Fraction of nominal mass flow rate where transition to turbulent occurs"
    annotation(Dialog(enable=use_deltaM));
- parameter Boolean use_v_nominal = true
-    "Set to true to use face velocity to compute area";
- parameter Modelica.SIunits.Velocity v_nominal=1 "Nominal face velocity"
```
Comment (mwetter) [inline Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo]: I suggest to leave this parameter, and use it below in `A=m_flow_nominal/rho_default/v_nominal`

File: Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo [hunk 2]
```diff
@@ -3,18 +3,14 @@ partial model PartialDamperExponential
...
-   annotation(Dialog(enable=use_v_nominal));
- parameter Modelica.SIunits.Area A=m_flow_nominal/rho_default/v_nominal
+ parameter Modelica.SIunits.Area A=m_flow_nominal/rho_default/1
```
Comment (mwetter) [inline Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo]: Use `v_nominal` rather than `1`.

File: Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo [hunk 3]
```diff
@@ -59,19 +55,19 @@ protected
...
  parameter Real facRouDuc= if roundDuct then sqrt(Modelica.Constants.pi)/2 else 1;
- parameter Modelica.SIunits.Area area=
-    if use_v_nominal then m_flow_nominal/rho_default/v_nominal else A
-    "Face velocity used in the computation";
 initial equation
   assert(k0 > k1, "k0 must be bigger than k1.");
   assert(m_flow_turbulent > 0, "m_flow_turbulent must be bigger than zero.");
+  assert(k1 >= 0.2, "k1 must be between 0.2 to 0.5.");
+  assert(k1 <= 0.5, "k1 must be between 0.2 to 0.5.");
+  assert(k0 <= 1e6, "k0 must be between k1 to 1e6.");
```
Comment (mwetter) [inline Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo]: Wrong text string

File: Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo [hunk 4]
```diff
@@ -126,6 +122,14 @@ Annex60.Fluid.Actuators.Dampers.Exponential</a>.
 revisions="<html>
 <ul>
 <li>
+October 12, 2016 by David Blum:<br/>
+Removed parameters <code>use_v_nominal</code> and <code>v_nominal</code>, 
+along with variable <code>area</code>, to simplify parameterization 
+of the model.  Also added assertion statements upon initialization
+for parameters <code>k0</code> and <code>k1</code> so that they fall within
+suggested ranges found in ASHRAE 825-RP.
```
Comment (mwetter) [inline Annex60/Fluid/Actuators/BaseClasses/PartialDamperExponential.mo]: Please add a link to the issue.

File: Annex60/Fluid/Actuators/Dampers/Examples/Damper.mo [hunk 5]
```diff
@@ -1,55 +1,70 @@
 within Annex60.Fluid.Actuators.Dampers.Examples;
 model Damper
-  "Damper with constant pressure difference and varying control signal"
+  "Dampers with constant pressure difference and varying control signal.  One damper uses a linear relation between mass flow and pressure drop."
```
Comment (mwetter) [inline Annex60/Fluid/Actuators/Dampers/Examples/Damper.mo]: Please use only one short sentence in this one-line command.

File: Annex60/Fluid/Actuators/Dampers/Examples/Damper.mo [hunk 6]
```diff
@@ -1,55 +1,70 @@
...
+    nPorts=2,
     p(displayUnit="Pa") = 101325,
     T=293.15)                                       annotation (Placement(
         transformation(extent={{74,10},{54,30}})));
 
+  Exponential resLinear(
+    redeclare package Medium = Medium,
+    m_flow_nominal=1,
+    filteredOpening=false,
+    linearized=true)
```
Comment (mwetter) [inline Annex60/Fluid/Actuators/Dampers/Examples/Damper.mo]: Comment is missing.

**Comment by dhblum** [comment]
I accepted the new results from the damper examples.  Examples.MixingBox and Examples.VAVBoxExponential each previously had use_v_nominal = true, despite also having damper areas specified by the parameter A.  For Examples.MixingBox A_out = 0.7, A_exh = A_rec = 1.0.  However, upon running the example, the variable used to compute flow was area_out = area_exh = area_rec = 0.83.  This was due to v_nominal and m_flow_nominal being equal for all dampers.  Similarly, for Examples.VAVBoxExponential, A_dam = A_vav = 1.8, however, when running the example area_dam = area_vav = 1.67.  

In summary, removing use_v_nominal in this issue has changed the damper areas used in the simulation to that specified as the parameter A.

**Comment by dhblum** [comment]
This change is not backwards compatible.  Any models using Dampers.Exponential would have to remove reference to the parameter use_v_nominal and be sure to properly specify the parameter A.

File: Annex60/Fluid/Actuators/Dampers/Examples/Damper.mo (hunk 6 above)
Comment (dhblum) [inline Annex60/Fluid/Actuators/Dampers/Examples/Damper.mo]: Comments for both damper model instantiations were added in commit 4600503.

**Comment by mwetter** [comment]
- [ ] The damper has `A`, `v_nominal` and `m_flow_nominal` as parameters. Remove to only use two.
- [ ] The same applies for the `MixingBox`

**Comment by mwetter** [comment]
Applied the following conversion script:
```
clear

convertClear();

convertModifiers("IBPSA.Fluid.Actuators.BaseClasses.PartialDamperExponential",
                 {"A"}, fill("",0), true);
convertModifiers("IBPSA.Fluid.Actuators.BaseClasses.PartialDamperExponential",
                 {"use_v_nominal"}, fill("",0), true);
convertModifiers("IBPSA.Fluid.Actuators.Dampers.MixingBox",
                 {"AOut", "AExh", "ARec"}, fill("",0), true);
convertModifiers("IBPSA.Fluid.Actuators.Dampers.MixingBox",
                 {"use_v_nominal"}, fill("",0), true);

openModel("package.mo");

```

**Comment by mwetter** [comment]
When merging, use "squash and merge" on merge button.

Library developers should check if they agree with the non-backward compatible test
- [ ] AixLib
- [ ] IDEAS
- [ ] BuildingSystems

**Comment by Mathadon** [comment]
ok for me!

**Comment by mwetter** [comment]
@Mathadon Can you please review and merge. Thanks. I reviewed (and revised) Dave's code, but it probably is better to not have people approve their co-workers' implementation.

**Comment by Mathadon** [comment]
Will do!
